src/JuMP.jl

```diff
@@ -77,7 +77,22 @@ An abstract type that should be subtyped for users creating JuMP extensions.
 """
 abstract type AbstractModel end
 
-mutable struct Model <: AbstractModel
+"""
+    value_type(::Type{<:Union{AbstractModel,AbstractVariableRef}})
+
+Return the return type of [`value`](@ref) for variables of that model. It
+defaults to `Float64` if it is not implemented.
+"""
+function value_type end
+
+value_type(::Type{<:AbstractModel}) = Float64
+
+"""
+    Model
+
+A mathematical model of an optimization problem.
+"""
```

Comment:
Please remove the docstring: https://github.com/jump-dev/JuMP.jl/pull/3198

Reply:
Oops, the rebases were a bit hard